statsmodels/genmod/generalized_estimating_equations.py

```diff
@@ -462,20 +462,20 @@ def unpack_cov(self, bcov):
 
 def _check_args(endog, exog, groups, time, offset, exposure):
 
-    if endog.size != exog.shape[0]:
+    if len(endog) != len(exog):
         raise ValueError("Leading dimension of 'exog' should match "
                          "length of 'endog'")
 
-    if groups.size != endog.size:
```

Comment:
These checks are not the same. size is the total number of elements. Len is the length of the 0-th axis. Ideally these checks should be corrected so check that the array_like is in factor 1-d and has the correct shape.  To really check this it should be a check something is 1-d and then the shapes match.  This requires using asarray to do it effectively.

Reply:
There are checks later again, but other `_check_inputs` are after the super call, i.e. after asarray.

Here the only check that was made, is checking the length, the code assumes everything except exog is  is 1-dim.
(checking `size` wasn't appropriate)

If we want to add ndim check, then the array conversion needs to happen in `__init__` to avoid unnecessary data memory duplication.

Reply:
> If we want to add ndim check, then the array conversion needs to happen in `__init__` to avoid unnecessary data memory duplication.

Or just convert and return them from this function.